server/common/fbs/matrix.py

```diff
@@ -132,7 +137,33 @@ def decode_matrix_fbs(fbs):
 
     columns_index = deserialize_typed_array((matrix.ColIndexType(), matrix.ColIndex()))
     if columns_index is None:
-        columns_index = range(0, n_cols)
+        columns_index = list(range(0, n_cols))
+    elif np.isscalar(columns_index):
+        columns_index = [columns_index]
+    elif isinstance(columns_index, np.ndarray) and columns_index.ndim == 0:
+        columns_index = [columns_index.item()]
+    elif isinstance(columns_index, range):
+        columns_index = list(columns_index)
+    else:
+        columns_index = list(columns_index)
+
+    def _normalize_column_key(key):
```

Comment:
decode_matrix_fbs: normalizes ColIndex (ranges/arrays/objects) via _normalize_column_key. Please check compatibility with any consumers that rely on previous key shapes.